clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2378,6 +2381,11 @@ public DescribeConsumerGroupsResult describeConsumerGroups(final Collection<Stri
 
         // TODO: KAFKA-6788, we should consider grouping the request per coordinator and send one request with a list of
         // all consumer groups this coordinator host
+
+        final Map<Integer, Queue<String>> coordinators = new ConcurrentHashMap<>();
+        final Map<Integer, FindCoordinatorResponse> fcResponses = new HashMap<>();
+        final Queue<String> returnedGroupId = new ConcurrentLinkedQueue<>();
+
```

Comment:
I suggest to use HashMap and LinkedList here since only one thread will process all responses.

Reply:
As I understand it, handleResponse will always be called by AdminClientRunnable from the single 'network thread' (KafkaAdminClient.thread).

Reply:
I see, I will modify it then. Thank you @andrasbeni @lambdaliu